pkg/sqlcache/informer/verify_generator_test.go

```diff
@@ -0,0 +1,667 @@
+			}, {
+				"project-01",
+				"project-04",
+			},
+		},
+	})
+	tests = append(tests, testCase{
+		description:   "external sort by country via foodCode, no filter, 4 per page",
+		queryTemplate: "page=%d&pagesize=4&sort=metadata.fields[2] => [_v1][Foods][foodCode][country]",
+		expectedResults: [][]string{
+			{
+				"cluster-01",
+				"cluster-02",
+				"cattle-pears",
+				"cattle-plums",
+			}, {
+				"cluster-bacon",
+				"cluster-eggs",
+				"project-03",
+				"fleet-local",
+			}, {
+				"local",
+				"project-01",
+				"cattle-lemons",
+				"default",
+			}, {
+				"before-local-01",
+				"project-04",
+				"project-05",
+				"cattle-mangoes",
+			}, {
+				"cattle-limes",
+				"fleet-default",
+				"cattle-kiwis",
+				"project-02",
+			},
+		},
+	})
+	tests = append(tests, testCase{
+		description:   "external sort active-only by country via foodCode, 7 per page",
+		queryTemplate: "page=%d&pagesize=7&filter=metadata.state.name=active&sort=metadata.fields[2] => [_v1][Foods][foodCode][country]",
+		expectedResults: [][]string{
+			{
+				"cluster-01",
+				"cattle-plums",
+				"fleet-local",
+				"cattle-lemons",
+				"default",
+				"before-local-01",
+				"project-05",
+			}, {
+				"cattle-mangoes",
+				"fleet-default",
+				"cattle-kiwis",
+				"project-02",
+			},
+		},
+	})
+
+	for _, test := range tests {
+		t.Run(test.description, func(t *testing.T) {
+			ctx := context.Background()
+			p := partition.Partition{Passthrough: true}
+			partitions := []partition.Partition{p}
+			namespace := ""
+			numTotal := 0
+			for _, part := range test.expectedResults {
+				numTotal += len(part)
+			}
+			for i, part := range test.expectedResults {
+				query := fmt.Sprintf(test.queryTemplate, i+1)
+				l, lo, err := getListOptionIndexerForQuery(t, ctx, query)
+				require.Nil(t, err)
+				queryInfo, err := l.constructQuery(lo, partitions, namespace, "_v1_Namespace")
+				require.Nil(t, err)
+				rows, err := func() (*sql.Rows, error) {
+					stmt, err := dbClient.Prepare(queryInfo.query)
+					require.Nil(t, err)
+					defer stmt.Close()
+					return stmt.Query(queryInfo.params...)
+				}()
+				require.Nil(t, err)
+				names, err := getFirstFieldFromRows(rows)
+				require.Nil(t, err)
+				assert.Equal(t, len(part), len(names))
+				assert.Equal(t, part, names)
+				if i == len(test.expectedResults)-1 {
+					// Verify that we've seen everything we expected to see
+					rows, err := func() (*sql.Rows, error) {
+						stmt, err := dbClient.Prepare(queryInfo.countQuery)
+						require.Nil(t, err)
+						defer stmt.Close()
+						return stmt.Query(queryInfo.countParams...)
+					}()
+					require.Nil(t, err)
+					var fullCount int
+					fullCount, err = getResultFromCountQuery(rows)
+					require.Nil(t, err)
+					assert.Equal(t, numTotal, fullCount)
+				}
+			}
+		})
+	}
+}
+
+func TestMain(m *testing.M) {
+	err := setupTests()
+	if err != nil {
+		panic(fmt.Sprintf("Awp! verify_generator_test.go tests failed to setup: %s", err))
```

Comment:
panic intententional?

Reply:
Given that `TestMain` doesn't return an error value, or anything, I assumed a panic was called for.